agent/php_newrelic.h

```diff
@@ -117,7 +117,7 @@ extern zend_module_entry newrelic_module_entry;
 
 #if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
     && !defined OVERWRITE_ZEND_EXECUTE_DATA
-#define NR_GET_RETURN_VALUE_PTR &func_return_value
+#define NR_GET_RETURN_VALUE_PTR func_return_value ? &func_return_value : NULL
```

Comment:
With this addition,[ all usages](https://github.com/newrelic/newrelic-php-agent/blob/9caf27b1bcd6ce58da28365b7a517180ad4a4d53/agent/fw_slim.c#L64) need to be doublechecked for null handling.

What is returned in case of an exception?  Is func_return_value null?  or is it pointing to a zval that has a value of NULL?

Reply:
I have manually checked all usages, though another pair of eyes could be useful. Furthermore, there are cases where the instrumentation is directly using `func_return_value` instead of calling `NR_GET_RETURN_VALUE_PTR`.

In the case of an exception, `func_return_value == NULL`

Reply:
doublechecked these cases:

```
agent/lib_predis.c
607:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/fw_laravel.c
1099:  route = NR_GET_RETURN_VALUE_PTR;

agent/fw_magento2.c
148:  response = NR_GET_RETURN_VALUE_PTR;
194:  retval_ptr = NR_GET_RETURN_VALUE_PTR;
294:  retval_ptr = NR_GET_RETURN_VALUE_PTR;
319:  retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/php_newrelic.h
120:#define NR_GET_RETURN_VALUE_PTR func_return_value ? &func_return_value : NULL
122:#define NR_GET_RETURN_VALUE_PTR nr_php_get_return_value_ptr(TSRMLS_C)

agent/fw_mediawiki.c
169:  return_value = NR_GET_RETURN_VALUE_PTR;

agent/fw_drupal8.c
211:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
268:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
367:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
401:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
551:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/fw_wordpress.c
626:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
690:    zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/lib_zend_http.c
341:  retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/fw_drupal.c
293:  if (NULL == NR_GET_RETURN_VALUE_PTR) {
310:        = nr_drupal_http_request_get_response_header(NR_GET_RETURN_VALUE_PTR);
313:        = nr_drupal_http_request_get_response_code(NR_GET_RETURN_VALUE_PTR);
395:  return_value = NR_GET_RETURN_VALUE_PTR;

agent/fw_slim.c
66:  retval_ptr = NR_GET_RETURN_VALUE_PTR;

agent/fw_laravel_queue.c
764:  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
```

all cases either explicitly check for NULL or the value is passed to a function that checks it for NULL.